### PR TITLE
Remove extra call to annotate function that messed up traces

### LINKED ISSUE #3560: Apex integration with HPX not working properly
## Expected Behavior
Nice plots of tasks

## Actual Behavior
All the tasks have the same `parallel::execution::post` name.
I am using the RAII style apex wrapper with `profiler::TaskProfiler tp(taskname, "Trailing Matrix");` that wraps an `hpx::util::annotate_function apex_profiler;` and is instantiated at the start of the task.
Note that is differs from the lambda style `annotated_function` by setting the name of the task _inside_ the task, whereas the lambda version does it on creation on the task.

I attempted to attach a screen shot of vampir, but it doesn't seem to work.

hoping @khuck knows what's wrong

### PR DESCRIPTION
**Pull request by biddisco**
Fixes #3560

### CODE REVIEW

File: hpx/util/annotated_function.hpp [hunk 1]
```diff
@@ -137,7 +137,6 @@ namespace hpx { namespace util
                 typename util::decay_unwrap<F>::type, Ts...>::type
             operator()(Ts && ... ts)
             {
-                annotate_function func(name_);
```
Comment (hkaiser) [inline hpx/util/annotated_function.hpp]: This call is the whole point of having the `annotated_function` type. Removing this call renders it useless.

Comment (khuck) [reply hpx/util/annotated_function.hpp]: @hkaiser What if we put `#if !defined(HPX_HAVE_APEX)` / `#endif` around it?  There is other code in HPX that will do the annotation correctly when the function happens to be of the type `annotated_function`.

Comment (biddisco) [reply hpx/util/annotated_function.hpp]: @hkaiser I'm not sure I understand. The `annotated_function` wrapper sets the annotation name in the creation of the task and does the 'right' thing. The call to annotate_function inside the `annotated_function` is doing what?

Comment (hkaiser) [reply hpx/util/annotated_function.hpp]: @biddisco right, I misunderstood - I think all is well now.

**Comment by biddisco** [comment]
I guess once the release is out, we should consider a point release in a month or so with some fixes like this in. A lot of people ask me about apex, and having a functional infrastructure would be a good thing.

**Comment by msimberg** [comment]
Thanks @biddisco for looking into this! Let's get this in the release as well. The annotation changes are very small. The ICE (https://cdash.cscs.ch//viewBuildError.php?buildid=91235) is a bit suspicious though as that test was actually changed... I've restarted that build so let's see if it fails again.

**Comment by msimberg** [comment]
This is definitely unrelated: https://cdash.cscs.ch//viewBuildError.php?buildid=91350.

**Comment by msimberg** [comment]
GCC 5 now builds cleanly, thanks @biddisco! https://cdash.cscs.ch/buildSummary.php?buildid=91377

Merging this now.

**Pull request merged**